core/roslib/src/roslib/packages.py

```diff
@@ -510,7 +509,9 @@ def find_resource(pkg, resource_name, filter_fn=None, rospack=None):
     for search_path in search_paths:
         matches.extend(_find_resource(search_path, resource_name, filter_fn=filter_fn))
 
-    matches.extend(_find_resource(pkg_path, resource_name, filter_fn=filter_fn))
+    # Only search ROS_PACKAGE_PATH if not matched in binary build dirs.
```

Comment:
This seems to be the opposite of the (old) lines 499-500. Please describe the use case where the current behavior is a problem and why it should be changed in this way.

Reply:
The current patch still contradicts the remaining comment. So this still needs changes one way or the other.

In general I am less inclined to change the code to match the docblock. Updating the docblock to match the implementation seems much safer.

If you think the behavior should be changed please describe the specific use case in detail in order to reproduce it. From the short comment above I am surprised that `am\ cl.launch` is returned when `amcl.launch` was looked for.